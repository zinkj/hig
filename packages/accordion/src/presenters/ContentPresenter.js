import React, { Component } from "react";
import PropTypes from "prop-types";
import { css, cx } from "emotion";
import { ThemeContext } from "@hig/theme-context";
import { createCustomClassNames } from "@hig/utils";

import stylesheet from "../stylesheet";

export default class ContentPresenter extends Component {
  static propTypes = {
    collapsed: PropTypes.bool,
    stylesheet: PropTypes.func,
    children: PropTypes.node
  };

  constructor(props) {
    super(props);
    this.contentInnerWrapper = null;
    this.setContentInnerWrapperRef = element => {
      this.contentInnerWrapper = element;
    };
  }

  render() {
    const {
      stylesheet: customStylesheet,
      collapsed,
      children,
      ...otherProps
    } = this.props;

    const { className } = otherProps;
    const contentTransitionWrapperClassName = createCustomClassNames(
      className,
      "content-transition-wrapper"
    );
    const contentInnerWrapperClassName = createCustomClassNames(
      className,
      "content-inner-wrapper"
    );

    return (
      <ThemeContext.Consumer>
        {({ resolvedRoles, metadata }) => {
          const styles = stylesheet(
            {
              collapsed,
              stylesheet: customStylesheet
            },
            resolvedRoles,
            metadata
          );

          return (
            <div
              className={cx(
                css(styles.contentTransitionWrapper),
                css({
                  height:
                    !collapsed && this.contentInnerWrapper
                      ? `${this.contentInnerWrapper.clientHeight}px`
                      : 0
                }),
                contentTransitionWrapperClassName
              )}
            >
              <div
                className={cx(
                  css(styles.contentInnerWrapper),
                  contentInnerWrapperClassName
                )}
                ref={this.setContentInnerWrapperRef}
              >
                {children}
              </div>
            </div>
          );
        }}
      </ThemeContext.Consumer>
    );
  }
}
