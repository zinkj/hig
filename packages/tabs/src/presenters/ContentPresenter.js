import React from "react";
import { cx, css } from "emotion";
import PropTypes from "prop-types";
import { createCustomClassNames } from "@hig/utils";
import stylesheet from "./ContentPresenter.stylesheet";

export default function ContentPresenter({ children, ...otherProps }) {
  const styles = stylesheet();
  const customClassName = createCustomClassNames(
    otherProps.className,
    "content"
  );

  return (
    <div className={cx(css(styles.content), customClassName)}>{children}</div>
  );
}

ContentPresenter.propTypes = {
  children: PropTypes.node
};
