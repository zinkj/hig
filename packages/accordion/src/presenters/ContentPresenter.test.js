import React from "react";
import renderer from "react-test-renderer";
import ContentPresenter from "./ContentPresenter";

describe("Accordion/ContentPresenter", () => {
  it("renders collapsed content", () => {
    const wrapper = <ContentPresenter collapsed />;
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders expanded content", () => {
    const wrapper = <ContentPresenter collapsed={false} />;
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
