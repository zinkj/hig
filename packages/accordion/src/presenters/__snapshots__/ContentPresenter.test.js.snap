// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion/ContentPresenter renders collapsed content 1`] = `
.emotion-1 {
  overflow: hidden;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  height: 0;
}

.emotion-0 {
  display: block;
}

<div
  className="emotion-1"
>
  <div
    className="emotion-0"
  />
</div>
`;

exports[`Accordion/ContentPresenter renders expanded content 1`] = `
.emotion-1 {
  overflow: hidden;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  height: 0;
}

.emotion-0 {
  display: block;
}

<div
  className="emotion-1"
>
  <div
    className="emotion-0"
  />
</div>
`;
